### PR TITLE
Additional conditionals

### PR DESCRIPTION
**Pull request by jesseweigert**
I added a couple of useful helpers: isLess and isBetween.

{#ifBetween total 100 1000}
10% discount applied
{/ifBetween}
{ifLess 1000 total}
20% discount applied
{/ifLess}

### CODE REVIEW

File: src/Comparison/IfLessHelper.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+namespace JustBlackBird\HandlebarsHelpers\Comparison;
+
+use Handlebars\Context;
+use Handlebars\Helper as HelperInterface;
+use Handlebars\Template;
+
+/**
+ * Conditional helper that checks if two values are equal or not.
```
Comment (JustBlackBird) [inline src/Comparison/IfLessHelper.php]: I think this comment is irrelevant to what the helper does.

File: src/Comparison/IfBetweenHelper.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+namespace JustBlackBird\HandlebarsHelpers\Comparison;
+
+use Handlebars\Context;
+use Handlebars\Helper as HelperInterface;
+use Handlebars\Template;
+
+/**
+ * Conditional helper that checks if two values are equal or not.
```
Comment (JustBlackBird) [inline src/Comparison/IfBetweenHelper.php]: I think this comment is irrelevant to what the helper does.

File: src/Comparison/IfBetweenHelper.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            throw new \InvalidArgumentException(
+                '"ifBetween" helper expects exactly three arguments.'
+            );
+        }
+
+        $variable = $context->get($parsed_args[0]);
+        $min = $context->get($parsed_args[1]);
+        $max = $context->get($parsed_args[2]);
+
+        $condition = ($variable >= $min && $variable < $max);
```
Comment (JustBlackBird) [inline src/Comparison/IfBetweenHelper.php:49]: This condition seems not obvious for me. I believe that _both_ conditions should be either strict or not strict (`min < value < max` or `min <= value <= max`). Also the type of conditions should be documented.

**Comment by JustBlackBird** [comment]
Thank you for the PR!

In general it looks good for me, but I've commented some problems in your commits.

File: src/Comparison/IfLessHelper.php (hunk 1 above)
Comment (jesseweigert) [inline src/Comparison/IfLessHelper.php]: Derp.. I may have copied that from another helper.  I'll fix it.

File: src/Comparison/IfBetweenHelper.php (hunk 3 above)
Comment (jesseweigert) [inline src/Comparison/IfBetweenHelper.php:49]: I can fix it, but I would like some input first.

The reason I wanted this conditional was to basically act as a big switch statement in templates.  I'm using this in an email campaign, so this kind of logic needs to be understandable by our marketing department.  In this case:

{#ifBetween total 0 100}$10 discount{/ifBetween}
{#ifBetween total 100 200}$15 discount{/ifBetween}
etc..

Maybe a different kind of construct could be done?

Comment (JustBlackBird) [inline src/Comparison/IfBetweenHelper.php:49]: Well, if we're talking about your use case I don't think that providing a discount for an order with zero total make any sense. So it should be like:

``` handlebars
{#ifEqual total 0}You should order something to get a discount{/ifEqual}
{#ifBetween total 0 100}$10 discount{/ifBetween}
{#ifBetween total 100 200}$15 discount{/ifBetween}
```

In such case `ifBetween` helper should be treated as a strict inequality (`min < value < max`).

Comment (JustBlackBird) [inline src/Comparison/IfBetweenHelper.php:49]: Sorry, I've missed the border values meaning. I should think for a while about possible usecases.

Comment (jesseweigert) [inline src/Comparison/IfBetweenHelper.php:49]: Maybe something like this?

```
{#range variable 0 100}
$10 discount
{#nextRange 200}
$15 discount
{#nextRange}
Big spender! $20 discount!
{/range}
```

Comment (JustBlackBird) [inline src/Comparison/IfBetweenHelper.php:49]: I don't think it's easily to implement in both Handlebars.php and Handlebars.js.

What do you think about using of several helpers? Ex.:
- `ifLess` stands for `(-inf, b)` (`var < b`)
- `ifBetween` stands for `(a, b)` (`a < var < b`)
- `ifBetweenLeftClosed` stands for `[a, b)` (`a <= var < b`)
- `ifBetweenRightClosed` stands for `(a, b]` (`a < var <= b`)
- `ifBetweenClosed` stands for `[a, b]` (`a <= var <= b`)
- `ifMore` stands for `(a, +inf)` (`var > a`)

The terminology is taken from math (see [wikipedia](https://en.wikipedia.org/wiki/Interval_%28mathematics%29#Classification_of_intervals)).

With the helpers above your use case can be written as:

``` handlebars
{#ifBetween var 0 100}
    $10 discount
{/ifBetween}
{#ifBetweenClosed var 100 200}
    $20 discount
{/ifBetweenClosed}
{#ifMore var 200}
    Big spender! $20 discount!
{/ifMore}
```

Comment (JustBlackBird) [inline src/Comparison/IfBetweenHelper.php:49]: Also we could use an abstract helper for all `ifBetween*` stuff. All child helpers only need to override a comparison method.

Comment (jesseweigert) [inline src/Comparison/IfBetweenHelper.php:49]: Sounds good.  I'll send another pull request later this week.  For now, I've got some more pressing issues at work(as usual), but I really want to see us switch from Smarty, and this is pretty much the only thing stopping us at this point. :-)

**Comment by JustBlackBird** [comment]
Sorry, I've implemented
- `ifLess`
- `ifBetween`
- `ifBetweenLeftClosed`
- `ifBetweenRightClosed`
- `ifBetweenClosed`
- `ifMore`

by myself. I'll release a new version of library soon.

**Comment by JustBlackBird** [comment]
Done. All the helpers available since [v1.1.0](https://github.com/JustBlackBird/handlebars.php-helpers/releases/tag/v1.1.0)

**Pull request closed without merging**